assets/product-form.js

```diff
@@ -35,23 +35,41 @@ if (!customElements.get('product-form')) {
         .then((response) => {
           if (response.status) {
             this.handleErrorMessage(response.description);
+
+            const soldOutMessage = this.submitButton.querySelector('.sold-out-message');
+            if (!soldOutMessage) return;
+            this.submitButton.setAttribute('aria-disabled', true);
+            this.submitButton.querySelector('span').classList.add('hidden');
+            soldOutMessage.classList.remove('hidden');
+            this.error = true;
             return;
           }
 
-          this.cartNotification.renderContents(response);
+          this.error = false;
+          const quickBuyModal = this.closest('quickbuy-modal');
+          if (quickBuyModal) quickBuyModal.hide(true);
+
+          if (!document.body.classList.contains('overflow-hidden')) {
+            this.cartNotification.renderContents(response);
+          } else {
+            document.body.addEventListener('modalClosed', () => {
+              setTimeout(() => { this.cartNotification.renderContents(response) });
+            }, { once: true });
+          }
```

Comment:
Could this be grouped in the `if (quickBuyModal)` or it wouldn't work 🤔 

So if `quickBuyModal` is `true`, then `hide` it and then have the `setTimeout()` ? Else `this.cartNotification.renderContents(response);`